### ISSUE TITLE
OpenAPI chopper stub generator

### ISSUE THREAD

**Issue by daniel-v** [lejard-h/chopper#99]
I'm just curious if this would make sense and gather some input on it from you guys. Dart OpenAPI client generators are lacking here and there. I'd be a nice feature for my projects to generate a chooper client library based on the spec.

How I imagine it would go:

1. grab spec file
2. run chopper client stub builder
3. run chooper builder

Maybe this package could help to read the specs file: `https://pub.dev/packages/open_api`.

I can see that `chopper_built_value` is under dev, it would play nicely together how I'm using it. `BuiltValue` based auto-generated, typed OpenAPI endpoints. Sounds good to me :)

**Comment by lejard-h** [comment]
Yes this is something I was planning to do a long time ago.
I started to work on it on a branch (https://github.com/lejard-h/chopper/tree/swagger) but very experimental and not up to date.

Generating Chopper client seems to be the easiest part, where it is more difficult it's about serialization, I tried to generate built_value class from an OpenAPI file but I find it difficult to understand, v2 or v3, all the different types, all the different kind of definitions. I am not familiar with openAPI, I have never use it on a real projects therefore I would accept any help.

About `chopper_built_value`, it's just about interfacing the `built_value` package with chopper not really about generating anything new. An issue was open but close now on built_value, maybe now is a good time to reopen it https://github.com/google/built_value.dart/issues/92
